framework/src/problems/FEProblemBase.C

```diff
@@ -177,6 +178,38 @@ FEProblemBase::validParams()
       "coord_type", coord_types, "Type of the coordinate system per block param");
   params.addParam<MooseEnum>(
       "rz_coord_axis", rz_coord_axis, "The rotation axis (X | Y) for axisymetric coordinates");
+  params.addParam<Real>("length_units_per_meter",
+                        "How many mesh length units are in a meter, e.g. if your mesh units are "
```

Comment:
let's have a default of 1?

Reply:
also this should be rangechecked

Reply:
actually should we have a non-uniform scaling operation instead of this?
It wont commute with the rotation anymore so we'd have to be pretty clear what's going on

Reply:
> let's have a default of 1?

I like being able to check `isParamValid`

> actually should we have a non-uniform scaling operation instead of this? It wont commute with the rotation anymore so we'd have to be pretty clear what's going on

How would you do this? Like with a `Function`?

Reply:
no as in instead of 1 have it '1 (x-direction scaling) 2 (y-direction scaling) 0.1 (z)'
I cant think of a use case rn though so dc too too much

Reply:
Why cannot this parameter a parameter of `MooseMesh` or `MooseApp`? I think this is a fairly important design decision. Also, if I set this parameter for cm, probably it won't work for tensor mechanics, how do you address that?

Reply:
It could live on `MooseApp` but we do not currently have a block for that in the input file. It could also live on `MooseMesh`, however, I think it makes sense to group these new parameters with coordinate information such as `coord_type` which is already in the `Problem` block

Reply:
Sounds a good argument. If it was in MooseApp, it will be set through command-line. I think a module can add an error if it never considers this parameter.

Reply:
Changed this to use the `MooseUnits` system